packages/flutter/lib/src/material/carousel.dart

```diff
@@ -296,6 +388,51 @@ class _CarouselViewState extends State<CarouselView> {
                     childCount: widget.children.length,
                   ),
                 ),
+                if (_weights != null) _SliverWeightedCarousel(
+                  allowFullyExpand: allowFullyExpand,
+                  shrinkExtent: widget.shrinkExtent,
+                  weights: _weights!,
+                  delegate: SliverChildBuilderDelegate(
+                    (BuildContext context, int index) {
+                      return Padding(
+                        padding: effectivePadding,
+                        child: Material(
+                          clipBehavior: Clip.antiAlias,
+                          color: effectiveBackgroundColor,
+                          elevation: effectiveElevation,
+                          shape: effectiveShape,
+                          child: Stack(
+                            fit: StackFit.expand,
+                            children: <Widget>[
+                              widget.children.elementAt(index),
```

Comment:
Slightly simplified:

```suggestion
                              widget.children[index],
```